includes/fbproductfeed.php

```diff
@@ -518,96 +326,133 @@ private function create_files_to_protect_product_feed_directory() {
+		 * @since x.x.x
+		 *
+		 * @throws Framework\SV_WC_Plugin_Exception We can't open the file or the file is not writable.
+		 * @return resource A file pointer resource.
+		 */
+		public function prepare_temporary_feed_file() {
+			$temp_file_path = $this->get_temp_file_path();
+			$temp_feed_file = @fopen( $temp_file_path, 'w' );
 
-					// skip if not enabled for sync
-					if ( ! Products::product_should_be_synced( $woo_product->woo_product ) ) {
-						continue;
-					}
+			// check if we can open the temporary feed file
+			if ( false === $temp_feed_file || ! is_writable( $temp_file_path ) ) {
+				throw new Framework\SV_WC_Plugin_Exception( __( 'Could not open the product catalog temporary feed file for writing', 'facebook-for-woocommerce' ), 500 );
+			}
 
-					$product_data_as_feed_row = $this->prepare_product_for_feed(
-						$woo_product,
-						$product_group_attribute_variants
-					);
+			$file_path = $this->get_file_path();
 
-					if ( ! empty( $temp_feed_file ) ) {
-						fwrite( $temp_feed_file, $product_data_as_feed_row );
-					}
-				}
+			// check if we will be able to write to the final feed file
+			if ( file_exists( $file_path ) && ! is_writable( $file_path ) ) {
+				throw new Framework\SV_WC_Plugin_Exception( __( 'Could not open the product catalog feed file for writing', 'facebook-for-woocommerce' ), 500 );
+			}
 
-				wp_reset_postdata();
+			fwrite( $temp_feed_file, $this->get_product_feed_header_row() );
+			return $temp_feed_file;
+		}
 
-				if ( ! empty( $temp_feed_file ) ) {
-					fclose( $temp_feed_file );
-				}
+		/**
+		 * Write products feed into a file.
+		 *
+		 * @since x.x.x
+		 *
+		 * @return void
+		 */
+		public function write_products_feed_to_temp_file( $wp_ids, $temp_feed_file ) {
+			$product_group_attribute_variants = array();
 
-				if ( ! empty( $temp_file_path ) && ! empty( $file_path ) && ! empty( $temp_feed_file ) ) {
+			foreach ( $wp_ids as $wp_id ) {
 
-					$renamed = rename( $temp_file_path, $file_path );
+				$woo_product = new WC_Facebook_Product( $wp_id );
 
-					if ( empty( $renamed ) ) {
-						throw new Framework\SV_WC_Plugin_Exception( __( 'Could not rename the product catalog feed file', 'facebook-for-woocommerce' ), 500 );
-					}
+				// Skip if we don't have a valid product object.
+				if ( ! $woo_product->woo_product instanceof \WC_Product ) {
+					continue;
 				}
 
-				$written = true;
+				// Skip if not enabled for sync.
+				if ( ! facebook_for_woocommerce()->get_product_sync_validator( $woo_product->woo_product )->passes_all_checks() ) {
+					continue;
+				}
 
-			} catch ( Exception $e ) {
+				$product_data_as_feed_row = $this->prepare_product_for_feed(
+					$woo_product,
+					$product_group_attribute_variants
+				);
 
-				WC_Facebookcommerce_Utils::log( json_encode( $e->getMessage() ) );
+				if ( ! empty( $temp_feed_file ) ) {
+					fwrite( $temp_feed_file, $product_data_as_feed_row );
+				}
+			}
 
-				$written = false;
+			wp_reset_postdata();
```

Comment:
What does this do here? Could you maybe add an inline comment describing why it's needed?